### DISCUSSION TITLE
Operator not creating masters component replicas

### DISCUSSION THREAD

**Discussion by nlcsdev** [Opster/opensearch-k8s-operator#351] [Q&A] [answered]
I started off following the steps of the guide and was able to spin up the dashboard and 3 cluster master pods, but at one point I deleted 2 of them via "kubectl delete" and changed the cluster master replica count to 1.

I haven't been able to make the operator spin up 3 cluster master since. I have tried deleting the cluster config and let the operator delete the resources, delete the operator, as well as clean out the persistent volume and persistent volume claims to ensure a fresh deployment, but none of that worked.

What else should I try?

**Comment by nlcsdev** [comment, marked as answer]
I have since resolved my issue. I deleted the configmaps generated from the deployment as well which seem to solve the issue.

**Reply by swoehrl-mw** [reply]
Hi @nlcsdev Good you know you managed to solve it on your own. Just wanted to add two points: Single-master clusters are not supported by the operator, so setting `replicas` to 1 will lead to undefined behaviour. And there is an open issue (#289) that deals with the operator currently being unable to recover from more than one pod being deleted/lost at the same time.
